src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -307,6 +307,10 @@ public override Binder VisitAccessorDeclaration(AccessorDeclarationSyntax parent
                         if ((object)accessor != null)
                         {
                             resultBinder = new InMethodBinder(accessor, resultBinder);
+                            if (propertyOrEventDecl.IsKind(SyntaxKind.PropertyDeclaration))
```

Comment:
>propertyOrEventDecl.IsKind(SyntaxKind.PropertyDeclaration) [](http://example.com/codeflow?start=32&length=58)

It looks like we can simply do a type check for ```SourcePropertyAccessorSymbol``` and keep the syntax check as an assert.

Reply:
@AlekseyTs Checking for `SourcePropertyAccessorSymbol` alone will include indexers. So if we want to get rid of syntactic check (which was actually done few lines above), then the symbol type check should include `IsIndexer` check as well. I'll do the symbol check and `IsIndexer` and adjust the syntactic checks as asserts.